src/WireMock.Net/RegularExpressions/RegexGuid.cs

```diff
@@ -0,0 +1,167 @@
+        /// `P` format specifier.
+        /// </summary>
+        public const string GuidPRegexPattern = @"(\([A-Z0-9]{8}-([A-Z0-9]{4}-){3}[A-Z0-9]{12}\))";
+
+        /// <summary>
+        /// Token for a GUID formated with `N` format specifier with lower case.
+        /// </summary>
+        public const string GuidNLowerToken = @"\guidn";
+
+        /// <summary>
+        /// Regular expression pattern associated with the expected format for
+        /// `N` format specifier with lower case.
+        /// </summary>
+        public const string GuidNLowerRegexPattern = "([a-z0-9]{32})";
+
+        /// <summary>
+        /// Token for a GUID formated with `N` format specifier.
+        /// </summary>
+        public const string GuidNToken = @"\GUIDN";
+
+        /// <summary>
+        /// Regular expression pattern associated with the expected format for
+        /// `N` format specifier.
+        /// </summary>
+        public const string GuidNRegexPattern = "([A-Z0-9]{32})";
+
+        /// <summary>
+        /// Token for a GUID formated with `X` format specifier with lower case.
+        /// </summary>
+        public const string GuidXLowerToken = @"\guidx";
+
+        /// <summary>
+        /// Regular expression pattern associated with the expected format for
+        /// `X` format specifier with lower case.
+        /// </summary>
+        public const string GuidXLowerRegexPattern = @"(\{0x[a-f0-9]{8},0x[a-f0-9]{4},0x[a-f0-9]{4},\{(0x[a-f0-9]{2},){7}(0x[a-f0-9]{2})\}\})";
+
+        /// <summary>
+        /// Token for a GUID formated with `X` format specifier.
+        /// </summary>
+        public const string GuidXToken = @"\GUIDX";
+
+        /// <summary>
+        /// Regular expression pattern associated with the expected format for
+        /// `X` format specifier.
+        /// </summary>
+        public const string GuidXRegexPattern = @"(\{0x[A-F0-9]{8},0x[A-F0-9]{4},0x[A-F0-9]{4},\{(0x[A-F0-9]{2},){7}(0x[A-F0-9]{2})\}\})";
+
+        /// <inheritdoc cref="Regex"/>
+        public RegexGuid(string pattern) : this(pattern, RegexOptions.None)
+        {
+        }
+
+        /// <inheritdoc cref="Regex"/>
+        public RegexGuid(string pattern,
+                         RegexOptions options)
+          : this(pattern, options, Regex.InfiniteMatchTimeout)
+        {
+        }
+
+        /// <inheritdoc cref="Regex"/>
+        public RegexGuid(string pattern,
+                         RegexOptions options,
+                         TimeSpan matchTimeout)
+          : base(ReplaceGuidPattern(pattern), options, matchTimeout)
+        {
+        }
+
+        /// <summary>
+        /// Replaces all instances of valid GUID tokens with the correct regular
+        /// expression to match.
+        /// </summary>
+        /// <param name="pattern">
+        /// Pattern to replace token for.
+        /// </param>
+        private static string ReplaceGuidPattern(string pattern)
```

Comment:
When using the dictionary, this can be simpler.
And a null check should be added.

``` c#
private static string ReplaceGuidPattern(string pattern)
{
    Check.NotNull(pattern, nameof(pattern));

     foreach (var replacement in _replacements)
     {
         pattern = pattern.Replace(replacement.Key, replacement.Value);
     }
}


```